modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java

```diff
@@ -1492,6 +1503,23 @@ private boolean isDeadNode(UUID uid) {
         return ctx.discovery().node(uid) == null || !ctx.discovery().pingNodeNoError(uid);
     }
 
+    /**
+     * <a href="https://issues.apache.org/jira/browse/IGNITE-6380">IGNITE-6380</a>
+     * restrict execution within transaction or locks if topology pending updates.
+     *
+     * @return {@code True} if execution must be restricted to prevent possible deadlock.
+     */
+    private boolean isRestrictExecution(ComputeJob job) {
```

Comment:
I think `isExecutionRestricted` or `isExecutionAllowed` better names for the method.